ocaml-eglot.el

```diff
@@ -33,6 +33,7 @@
 ;;; Code:
 
 (require 'flymake)
+(require 'flycheck)
```

Comment:
I don't think it's a good idea to require flycheck here, as it's not installed by default. If someone's using flycheck it will likely be already loaded by the time `ocaml-eglot` gets loaded, so you can just declare whatever functions from it are going to be used here to silence the byte-compiler.

Reply:
I'm not (yet!) really familiar with scope presence and require.
What's the best way to avoid relying on `flycheck` by default? Is it possible, at the `ocaml-eglot` level to behave differently if `flycheck` is installed and ideally to rely on additional behavior in case of `flycheck-eglot`?

Reply:
Normally you'd use `fboundp` to check if something is present at runtime (e.g. as in https://github.com/clojure-emacs/cider/blob/6594b4aa26a6e4255bd5d7028221130d35619b2c/cider-log.el#L118)

`require` itself can be called from any place in an Elisp package (e.g. when a certain function is triggered), or it can be called with optional parameters that will make it just return `nil` is some package you're trying to require is not present.